lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -698,4 +781,47 @@ public static function get_style_variations() {
 		return $variations;
 	}
 
+	/**
+	 * Returns all style variations.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @return array
+	 */
+	public static function get_user_style_variations() {
```

Comment:
This method is the one used by the REST endpoint, and it returns the user style variations to be shown in the UI. I see it returns all `wp_global_styles` CPT: the "user styles" and the "user style variations". Should the "user styles" be listed in the "custom style variations" UI?

Reply:
That's the only place where this method is used currently, so in that sense, no, we shouldn't return CUGS from this method. It actually causes more JS work later (which I've done) because CUGS needs to be filtered out.

But when I was writing this method, I was thinking of it as something that might be used in general to get all user global styles, without worrying about the arguments that need to be passed to WP_Query. Perhaps it will be used by plugins or other WP core places.

I'm not sure what the right answer is here. When I first wrote this method I didn't include CUGS, later I reverted because I was like... Well, the REST endpoint to get all global styles should actually get all global styles. That's how it works for posts and other objects. So, I'm not sure. If you think I should exclude CUGS, let me know and I'll do it.